src/libsync/syncengine.cpp

```diff
@@ -912,7 +912,16 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
             return;
         }
     }
-    if (!_hasForwardInTimeFiles && _backInTimeFiles >= 2) {
+
+    auto databaseFingerprint = _journal->dataFingerprint();
+    // If databaseFingerprint is null, this means that there was no information in the database
+    // (for example, upgrading from a previous version, or first sync)
+    // If it is empty (""), then that meanst that it was empty on the server before.
```

Comment:
no, empty is not isNull   hence the isNull   which is different from isEmpty because the isEmpty needs to be covered by the normal case.
My comment is just there so someone does not change isNull to isEmpty,  or at least understands why it is isNull and not isEmpty